### PR TITLE
Planned Enhancements for Event

### LINKED ISSUE #201: [PE-D][Tester D] Event is case-sensitive
Able to add duplicated event with different case-sensitivity when events like "birthday" refer to the same event

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: YowSiaoKang/ped#8

### LINKED ISSUE #207: [PE-D][Tester E] Date for event can be 4000 years into the future.
## Description

I am able to add a date 4000 years into the future, which by the time, the patient would have already died.

## Fix

Create an upper and lower bound for dates.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: c-wenlong/ped#10

### LINKED ISSUE #233: [PE-D][Tester A] Inaccurate Date check
As shown, 30th of Feburary should not be possible on the calendar.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: AL-ZT/ped#11

### PR DESCRIPTION
**Pull request by SherisseTJW**
Add planned enhancements for Event related issues.

Fixes #233
Fixes #207 
Fixes #201 

Also, fixed the error message for Event to standardise with the UG. Approval was sought and obtained, see: https://github.com/nus-cs2103-AY2324S2/forum/issues/791

Lastly, added a note to the UG about the state of the data for Screenshots

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -922,6 +922,27 @@ Currently, the process of modifying patient tags in the PatientSync application
 To address this, we are planning to introduce the EditTagsCommand feature, providing users with a more flexible and efficient way to manage patient tags. The planned EditTagsCommand feature is designed to enhance user productivity, reduce the likelihood of errors, and improve overall usability within PatientSync.
 
 
+### 4.3 Input Validation for Events
+
+Presently, the Date and Datetime for Events, referred to as `DATE_OR_DATETIME_OF_EVENT`, do not have sufficient input validation. For example, the user is able to input `30-02-2024, 24:00 - 24:00`. This can lead to potential errors if the user has accidentally mistyped the date and/or time when inputting the command, leading to confusion further down the line.
```
Comment (ZhiWei1010) [inline docs/DeveloperGuide.md]: Perhaps including the current input validation present, and planned specific input validation such as invalid time (eg. 24:00) or invalid dates (eg 31-02-2024) would be clearer!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -922,6 +922,27 @@ Currently, the process of modifying patient tags in the PatientSync application
...
+
+To address this, we plan to make our input validation for the `DATE_OR_DATETIME_OF_EVENT` field stricter, to ensure the validity of the values, and not just the format. Upon identification of such invalid `DATE_OR_DATETIME_OF_EVENT` field values, PatientSync should then output a custom error message, i.e.,`Invalid DATE_OR_DATETIME_OF_EVENT!`
+
+
+### 4.4 Addition of an Upper and Lower Bound for Event Date or Datetime
+
+Presently, we do not restrict the user from adding events at any date. As such, the user is able to add Events for a Patient in impossible dates, i.e., 4000 years into the Future or Past. Adding such Events is more likely to be a typographical error and thus, we should warn the user (similar to how we warn the user for past events). We choose to warn the user rather than error on the command as it is possible, albeit rare, for a Person to live more than a 100 years.
```
Comment (ZhiWei1010) [inline docs/DeveloperGuide.md]: Perhaps could change ~impossible dates~ unrealistic

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1155,7 +1176,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 2. Performance: Should be able to hold up to 1000 patients without a noticeable sluggishness in performance for typical usage.
 3. Usability: A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 4. Accessibility: Should operate without the need for internet access to fulfill its core purpose.
-5. Ease of Use: Should be designed to be usable by a patient new to patient management without extensive training.
+5. Ease of Use: Should be designed to be usable by a nurse new to patient management without extensive training.
```
Comment (ZhiWei1010) [inline docs/DeveloperGuide.md:1186]: good catch!

File: docs/UserGuide.md [hunk 4]
```diff
@@ -54,6 +54,10 @@ command to run the application.<br>
 
 <box type="info" seamless>
 
+**Notes about the Screenshots:**<br>
+
+* Each Screenshot is taken based on the state of the data after running the command example stated. The original state of the data before running the command is that of the base Sample Data provided.
```
Comment (ZhiWei1010) [inline docs/UserGuide.md]: maybe could add: based on the **original** state of data
~command example~ -> example command  
The original state of the data before running the command is that of the base Sample Data provided.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -922,6 +922,34 @@ Currently, the process of modifying patient tags in the PatientSync application
 To address this, we are planning to introduce the EditTagsCommand feature, providing users with a more flexible and efficient way to manage patient tags. The planned EditTagsCommand feature is designed to enhance user productivity, reduce the likelihood of errors, and improve overall usability within PatientSync.
 
 
+### 4.3 Input Validation for Events
+
+Presently, the Date and Datetime for Events, referred to as `DATE_OR_DATETIME_OF_EVENT`, do not have sufficient input validation. For example, the user is currently able to input `30-02-2024, 24:00 - 24:00`. However, `30-02-2024` is not a valid Date and `24:00` is not a valid time.
```
Comment (ZhiWei1010) [inline docs/DeveloperGuide.md]: Just to check: are the current input validation implemented already mentioned in the Events section of the DG? if not, I suggest could include the current validation checks performed!

Comment (SherisseTJW) [reply docs/DeveloperGuide.md]: not validated properly;

Its using the LocalDate and LocalTime but those allow for these invalid dates

**Review by ZhiWei1010 — APPROVED** [review]
LGTM!

**Pull request merged**